src/vmath.nim

```diff
@@ -507,44 +507,32 @@ type Mat3* = array[9, float32] ## 3x3 Matrix
 template `[]`*(a: Mat3, i, j: int): float32 = a[i * 3 + j]
 template `[]=`*(a: Mat3, i, j: int, v: float32) = a[i * 3 + j] = v
 
-proc mat3*(a, b, c, d, e, f, g, h, i: float32): Mat3 =
-  result[0] = a
-  result[1] = b
-  result[2] = c
-  result[3] = d
-  result[4] = e
-  result[5] = f
-  result[6] = g
-  result[7] = h
-  result[8] = i
-
-proc mat3*(a: Mat3): Mat3 =
+proc mat3*(a, b, c, d, e, f, g, h, i: float32): Mat3 {.inline.} =
+  [
+    a, b, c,
+    d, e, f,
+    g, h, i
+  ]
+
+proc mat3*(a: Mat3): Mat3 {.inline.} =
   a
 
-proc identity*(a: var Mat3) =
-  a[0] = 1
-  a[1] = 0
-  a[2] = 0
-  a[3] = 0
-  a[4] = 1
-  a[5] = 0
-  a[6] = 0
-  a[7] = 0
-  a[8] = 1
+proc identity*(a: var Mat3) {.inline.} =
+  a = [
+    1.float32, 0, 0,
+    0,         1, 0,
+    0,         0, 1
+  ]
 
 proc mat3*(): Mat3 {.inline.} =
   result.identity()
 
-proc transpose*(a: Mat3): Mat3 =
-  result[0] = a[0]
-  result[1] = a[3]
-  result[2] = a[6]
-  result[3] = a[1]
-  result[4] = a[4]
-  result[5] = a[7]
-  result[6] = a[2]
-  result[7] = a[5]
-  result[8] = a[8]
+proc transpose*(a: Mat3): Mat3 {.inline.} =
+  [
+    a[0], a[3], a[6],
```

Comment:
I think using the new [0, 0] would make this even more clear.

Reply:
But it can be done later...

Reply:
benchmarked using them, makes no perf diff so switched